org-tanglesync.el

```diff
@@ -302,13 +302,19 @@ Only takes effect when :custom is set"
   (interactive)
   (org-tanglesync-process-buffer t))
 
+(defun tangled-p ()
+  "Check if :tangle property exists"
+  (let* ((properties (car (cdr (cdr (org-babel-get-src-block-info))))))
+    (if (string= (cdr (assoc :tangle properties)) "no")
+        nil 't)))
+
 (defun org-tanglesync-user-edit-buffer ()
   "A hook to the org-src-code mode.  If there is an external change, prompt the user unless the `org-tanglesync-skip-user-check` custom parameter is set."
   (let* ((edit-buffer (current-buffer))
          (org-buffer (org-src-source-buffer))
          (org-position org-src--beg-marker))
     (with-current-buffer org-buffer
-      (when (bound-and-true-p org-tanglesync-mode)
+      (when (and (bound-and-true-p org-tanglesync-mode) (tangled-p))
         (goto-char org-position)
         (let* ((tangle-fname (org-tanglesync-get-tangledfile
```

Comment:
this line is where it tests that the block has a tangle specified (`tangle-fname`) and should return `nil` if none can be found, so I am wondering why you say this fails....

Reply:
I came to that by trial and error i.e. setting `(setq debug-on-error t)` and noticed that the error being thrown was somewhere in that block. My line of reasoning is that as you check whether `org-tanglesync-mode` is bound, you should also check whether the `tangle` property exists. Anyway, this is irrelevant since the fix you posted works.

Reply:
Oh nice, does debug on error allow you to step through the code?

I normally just pull up the source file from the elpa directory, eval the buffer, and then for the function I want to debug I press C-u C-M-x, and then when the function is called via hook or whatnot, it let's me step through each statement via SPC.